yt/visualization/plot_window.py

```diff
@@ -1118,6 +1142,14 @@ def _setup_plots(self):
                 ia = ImageArray(ia)
             else:
                 ia = image
+
+            swap_axes = self._swap_axes
+            aspect = self.aspect
+            if swap_axes:
+                extent = [extent[i] for i in (2, 3, 0, 1)]
+                ia = ia.transpose()
+                aspect = 1.0 / aspect
```

Comment:
this is almost definetely going to collide with #3633, I think we need to discuss what `self.aspect` exactly means or should mean.

Reply:
ya, I wasn't even entirely sure what `aspect` does here -- assumed it was an aspect ratio of some kind and so it should also be inverted. So yes, this needs more investigation and discussion.

Reply:
oh, this `aspect` is the `imshow(aspect=` keyword arg... it's a string, either `equal` or `auto` . The actual aspect ratio which we call `_unit_aspect` is calculated from `extent` within `WindowPlotMPL.__init__`. So I shouldn't be touching `aspect` here at all.

Reply:
oops, maybe I was wrong about that.... but ya, this needs to be sorted out...

Reply:
ok, missed that aspect could be "{'equal', 'auto'} or float" to `imshow`, and we set `aspect=1.` earlier on if it is `None`. So in isolation what I did is right, but ya, need to review it in light of #3633

Reply:
in #3633 I've assumed that the `_unit_aspect` ratio should always be read as vertical/horizontal. I'm open to debate if you feel there's a more intuitive interpretation on the user side.

Reply:
| `_unit_aspect` ratio should always be read as vertical/horizontal.

So I'm still not positive on this, but I think my feeling is that if the aspect ratio is set and then I  call `slc.swap_axes()` then I would want the aspect ratio to flip as well? i.e., I would find it more intuitive if the following resulted in two images with different aspect ratios:

```
slc = yt.SlicePlot(ds, "z", "density", aspect=0.2)
slc.show()
slc.swap_axes()
slc.show()
```

Though maybe it gets more confusing in the case of setting both in the initial plot call. Because I think I would expect `yt.SlicePlot(ds, "z", "density", aspect=0.2, swap_axes=True)` to NOT invert the aspect ratio...